hub/templates/hub/tools/postcode.html

```diff
@@ -0,0 +1,169 @@
+            // if trackEvent is defined, log the number of postcodes processed
+            if (typeof trackEvent === "function") {
+                    trackEvent("postcode_tool", {"count": numPostcodes});
+            }
+        }
+
+        function process() {
+            var output_type = document.querySelector('input[name="output_type"]:checked').value;
+
+            var postcodes = document.getElementById("postcodes").value.split("\n");
+            var constituencies = postcodes.map(postcode => lookup.getValue(postcode));
+
+            // if output_type is not mySoc ID, convert the constituency ids to the desired output
+            if (output_type !== "mysoc_id") {
+                constituencies = constituencies.map(constituency => {
+                    if (constituency === null) {
+                        return null;
+                    }
+                    return field_lookup[constituency][output_type];
+                });
+            }
+
+            // if the top row of constituencies is blank
+            // there is more than one row (suggesting the original top row was a header)
+            // make the top row a pcon_2024 header
+            if (constituencies[0] === null && constituencies.length > 1) {
+                constituencies[0] = "pcon_2024";
+            }
+
+            document.getElementById("constituencies").value = constituencies.join("\n");
+
+            log_count(postcodes.length);
+        }
+
+
+        document.getElementById("postcodes").addEventListener("input", process);
+        document.getElementsByName("output_type").forEach(function (element) {
+            element.addEventListener("input", process);
+        });
+
+        document.getElementById("copyButton").addEventListener("click", function () {
+            var constituenciesText = document.getElementById("constituencies").value;
+            navigator.clipboard.writeText(constituenciesText)
+                .then(() => {
+                    console.log("Constituencies copied to clipboard");
+                })
+                .catch(error => {
+                    console.error("Error copying constituencies to clipboard:", error);
+                });
+            // change the copyButton class to btn-success and its text to 'Copied to clipboard'
+            // for two seconds before reverting to the original class and text
+            document.getElementById("copyButton").classList.remove("btn-primary");
+            document.getElementById("copyButton").classList.add("btn-success");
+            document.getElementById("copyButton").textContent = "Copied to clipboard";
+            setTimeout(function () {
+                document.getElementById("copyButton").classList.remove("btn-success");
+                document.getElementById("copyButton").classList.add("btn-primary");
+                document.getElementById("copyButton").textContent = "Copy Constituencies to clipboard";
+            }, 500);
```

Comment:
This does not seem like an obvious way to do this. I suspect having a message that says "copied to the clipboard" which is then removed if they add another postcode.

Reply:
Yep, that's nicer